### DISCUSSION TITLE
Challenge lesson 8, can't access

### DISCUSSION THREAD

**Discussion by usereri** [Cyfrin/foundry-full-course-cu#1577] [Q&A] [answered]
Hi all, 
I am trying to get my hands on challenge from lesson 8 but it is not opening (the link for ipfs site is not working). I tried to get the contract address from NFT factory contract but when I try to enter index 7 to getChallenge function it returns not published/verified contract which doesn't help :( 

Could anyone tell me how you approached this challenge? 

Thanks in advance! 
Eri

**Comment by EngrPips** [comment, marked as answer]
Hello @usereri , after i carefully read and study the transactions that have been sent to the contract address of the lesson 8 challenge, i realize that you just have to call the `solveChallenge` method on the contract address `[0xf215a0b6DD88d6029b5385D6fab51968337E963D]` of the challenge. Yea the contract isn't verified on etherscan so you won't have the luxury of just calling the method by clicking on some button on a user interface. So you would have to do a low level call by using `contractAddress.call(callData)` the `solveChallenge` method takes two argument as shown below 

<img width="1391" alt="Screenshot 2024-04-05 at 2 41 56 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/ac16c198-65ec-45a6-9ab4-cbd89ac48768">

As you can see in the above image the first argument is a uint256 and the second argument is of type string which is your twitter handle. I have noticed every person that successfully call the `solveChallenge` method on the contract address `[0xf215a0b6DD88d6029b5385D6fab51968337E963D]` have always passed in `123456789` as the uint256 value and their twitter handle as the string. You can send the transaction using a foundry script if you will. 

I hope you have hint on about how to solve the challenge now

**Comment by blockcipherdev** [comment]
The contract address in the constants.js file needs to be updated to 0xf215a0b6DD88d6029b5385D6fab51968337E963D.

https://ipfs.io/ipfs/QmUgc1xpNgTMwSF2GBS5FTFKmSdrmSnLJD5DoYZtcQ2n5X/ -> constants.js points to a different smart contract. Gas estimation will always fail.

You can download the files locally, edit the sepoliaAddress to 0xf215a0b6DD88d6029b5385D6fab51968337E963D and run on a local server. 

Fun fact: Index for this challenge (Lesson 8) is 11. :)

**Reply by EngrPips** [reply]
Wow seriously?

**Reply by usereri** [reply]
I was able to solve that challenge through cast module in foundry.

**Reply by EngrPips** [reply]
Interesting

**Reply by blockcipherdev** [reply]
As per my understanding, the lesson objective is to call the contract function using a HTML front end.

**Comment by blockcipherdev** [comment]
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/104627284/678de409-7452-4d84-bcfb-5e1399d5002b)

**Reply by EngrPips** [reply]
Hello @blockcipherdev, Do you have any problem at all here?
